File: medplum-java/medplum-java-model/src/main/java/com/fhir/Consent_Provision.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.String;
import java.util.List;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutableConsent_Provision.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface Consent_Provision {
  Optional<Period> period();

  Optional<List<Extension>> extension();

  Optional<List<Coding>> securityLabel();

  Optional<List<Consent_Actor>> actor();

  Optional<Period> dataPeriod();

  @JsonProperty("class")
  Optional<List<Coding>> _class();

  Optional<List<CodeableConcept>> code();

  Optional<List<Consent_Provision>> provision();

  Optional<List<Extension>> modifierExtension();

  Optional<List<Consent_Data>> data();

  Optional<List<CodeableConcept>> action();

  Optional<String> id();

  Optional<List<Coding>> purpose();

  Optional<Consent_provisionType> type();

  static ImmutableConsent_Provision.Builder builder() {
    return ImmutableConsent_Provision.builder();
  }
}

File: medplum-typescript/src/model/fhir/Consent_Provision.ts
import { CodeableConcept } from "./CodeableConcept";
import { Coding } from "./Coding";
import { Consent_Actor } from "./Consent_Actor";
import { Consent_Data } from "./Consent_Data";
import { Consent_Provision } from "./Consent_Provision";
import { Consent_provisionType } from "./Consent_provisionType";
import { Extension } from "./Extension";
import { Period } from "./Period";

export interface Consent_Provision {
    id: string | null | undefined;
    extension: Extension[] | null | undefined;
    modifierExtension: Extension[] | null | undefined;
    type: Consent_provisionType | null | undefined;
    period: Period | null | undefined;
    actor: Consent_Actor[] | null | undefined;
    action: CodeableConcept[] | null | undefined;
    securityLabel: Coding[] | null | undefined;
    purpose: Coding[] | null | undefined;
    class: Coding[] | null | undefined;
    code: CodeableConcept[] | null | undefined;
    dataPeriod: Period | null | undefined;
    data: Consent_Data[] | null | undefined;
    provision: Consent_Provision[] | null | undefined;
}
